src/FastConv/FastConv.cpp

```diff
@@ -1,289 +1,32 @@
-            {
-                complexMultiply(pfRealInputProcessing,pfImagInputProcessing,ppfRealBlockedIR[j],ppfImagBlockedIR[j],pfProductReal,pfProductImag,m_lengthofIR);
-                m_pCFft->mergeRealImag(pFFTProductProcess,pfProductReal,pfProductImag);
-                m_pCFft->doInvFft(pfInvFFtProcessing,pFFTProductProcess);
-                int process_writeIdx = (BlockNoWriting + j) % numOfIRBlocks; //A temporary variable to help write the convolution sum with each block of the IR
-                for (int k =0; k<m_BlockLength;k++)
-                {
-                    ppfProcessedOutputBlocks[process_writeIdx][k] += pfInvFFtProcessing[k+m_BlockLength];
-                }
-            }
-            BlockNoReading=BlockNoWriting;
-            //Shift the current input block to the left to make room for the new x2
-            for (int k = 0; k<m_BlockLength;k++)
-            {
-                pfInputProcessing[k] = pfInputProcessing[k+m_BlockLength];
-            }
-            //Make the 2 dimensional processblocks behave like a ringbuffer (Can just use the ringbuffer to control these shifts)
-
-            BlockNoWriting=(BlockNoWriting+1)%numOfIRBlocks;
-        }
-
-    }
-
-
-
-
-    return Error_t::kNoError;
-}
-
-//FUNCTION TO MAKE COMPLEX MULTIPLICAION, GIVEN COMPLEX_T
-//splitRealImag
-
-Error_t CFastConv::complexMultiply(float* realInput1, float* imagInput1, float* realInput2, float* imagInput2, float* realOutput, float* imagOutput, int outputLength ) {
-    for (int i = 0; i < outputLength; i++) {
-        realOutput[i] = (realInput1[i] * realInput2[i]) - (imagInput1[i] * imagInput2[i]);
-        imagOutput[i] = (realInput2[i] * imagInput1[i]) + (realInput1[i] * imagInput2[i]);
-    }
-    return Error_t::kNoError;
-}
-
-
-//TODO: [10] Implement the 'flushBuffer' function that will return the remaining result (reverb tail) after the last sample has been processed (end of input signal).
-// flushBuffer is to be called once after the last process call. Question: How will you make it clear to the user what amount of memory to allocate for the output?
-// Do you need an additional API function?
-
 Error_t CFastConv::flushBuffer(float* pfOutputBuffer)
 {
-    if(type == kTimeDomain)
-    {
-        float *pfInputFlush = new float[m_lengthofIR - 1];
-        long long int sbufferlength = m_lengthofIR-1;
-        CVectorFloat::setZero(pfInputFlush, sbufferlength);
-        process(pfOutputBuffer, pfInputFlush, m_lengthofIR - 1);
-        delete[] pfInputFlush;
-    }
-    if(type == kFreqDomain)
-    {
-        float *pfInputFlush = new float[m_BlockLength + m_lengthofIR - 1]; //Think about the initial delay of the zeros of blocksize
-        long long int sbufferlength = m_lengthofIR + m_BlockLength-1;
-        CVectorFloat::setZero(pfInputFlush, sbufferlength);
-        process(pfOutputBuffer, pfInputFlush, m_BlockLength+m_lengthofIR - 1);
-        delete[] pfInputFlush;
-
-    }
```

Comment:
For this function, I would suggest not even doing any processing whatsoever. Perhaps you could create a member float buffer like `float* m_pfTail = new float[<whatever the tail length is>]` then store all leftover values into this buffer during your main process function. Then, when it comes time to flush the buffer, simply copy over the values into `pfOutputBuffer` by doing something like `CVectorFloat::copy(pfOutputBuffer, m_pfTail, iLengthOfTail)`